libs/blocks/global-navigation/global-navigation.js

```diff
@@ -621,10 +621,12 @@ class Gnav {
       children: getChildren(),
     });
 
-    window.UniversalNav(getConfiguration());
+    // Exposing UNAV config for consumers
+    CONFIG.universalNav.universalNavConfig = getConfiguration();
+    window.UniversalNav(CONFIG.universalNav.universalNavConfig);
 
     isDesktop.addEventListener('change', () => {
-      window.UniversalNav.reload(getConfiguration());
+      window.UniversalNav.reload(CONFIG.universalNav.universalNavConfig);
```

Comment:
I'm not very familiar with this logic, but judging by its structure, it seems to indicate that the config could be different between viewports. This approach would just re-use the initial config. You could dispatch some new events when the config changes or define a new method in this file that handles all of the reload complexities. Consumers would then use that method instead of the regular unav one.

Reply:
@overmyheadandbody It seems as though `getConfiguration` returns the same thing every time:
```javascript
    const getConfiguration = () => ({
      target: this.blocks.universalNav,
      env: environment,
      locale,
      imsClientId: window.adobeid?.client_id,
      theme: 'light',
      onReady: () => {
        this.decorateAppPrompt({ getAnchorState: () => window.UniversalNav.getComponent?.('app-switcher') });
      },
      analyticsContext: {
        consumer: {
          name: 'adobecom',
          version: '1.0',
          platform: 'Web',
          device: getDevice(),
          os_version: navigator.platform,
        },
        event: { visitor_guid: visitorGuid },
        onAnalyticsEvent,
      },
      children: getChildren(),
    });
```
It seems that we simply call `getConfiguration` again because the unav's reload method requires it. But if modifications have been made to the original configurations, we don't want those to be lost if the viewport is changed.